mockgcp/pkg/validation/labels.go

```diff
@@ -0,0 +1,52 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package validation
+
+import (
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+// ValidateLabels checks if a map of labels conforms to GCP's label requirements.
+func ValidateLabels(labels map[string]string) error {
```

Comment:
Maybe start by adding this by hand to a few mockgcp resources?  Now we have AI, I value clarity over fewer lines of code

Reply:
2 GCP services use the `labels` field for a different purpose. They are caught by this change. Please see the last git-commit